var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -1104,3 +1113,16 @@ def detect_gdc(self):
                         ),
                     ),
                 )
+
+    @property
+    def command(self):
+        """Returns the name of the main compiler executable.
+
+        If the program_suffix variant was specified, append the suffix to the
+        ``gcc`` string.
+        """
+        spec = self.spec
+        program_suffix = spec.variants["program_suffix"].value
+        if program_suffix == "none":
+            program_suffix = ""
+        return f"gcc{program_suffix}"
```

Comment:
You unfortunately can't assume that `command` isn't going to be called on an older installation of the package that doesn't have the variant.

```suggestion
        program_suffix = spec.variants.get("program_suffix")
        suffix = program_suffix.value if program_suffix and program_suffix.value != "none" else ""
        return f"gcc{suffix}"
```